Template formatted output for ps & inspect

darkowlzz opened the pull request:
Adds new flag `--template` (`-t`) for Go template formatted output in `inspect` and `ps` subcommands.
Since `ps` command already has a `-f` flag for `--filter`, this uses `-t` for template formatted output.

Adds golden files based tests for Inspect() and Ps() to verify the results.

darkowlzz commented:
Updated the commits to replace separate flags for updating the test golden files with a single `-update` flag for all the test updates.

stealthybox commented:
The template format works great, but the golden-file tests are flaky because the output order is not sorted and stable.

Running with `-count 1` will disable test caching.
Try `-count 10` as well:
```
go test ./cmd/ignite/run -count 10 -run TestPs
```

Perhaps we should consider sorting by name or ID before outputting the table.
We don't need to fix this in this PR.

darkowlzz commented:
Thanks for catching the issue.
Rebased the changes on latest master.
Changed the list of existing VMs to use a slice of `runtime.ObjectMeta`, instead of a map of VM name and ID, and tested it with `-count=10`. Seems to work reliably now.
